packages/block-library/src/image/edit.js

```diff
@@ -340,10 +340,15 @@ export class ImageEdit extends Component {
 		if ( currentRel !== undefined && ! isEmpty( newRel ) ) {
 			if ( ! isEmpty( newRel ) ) {
 				each( NEW_TAB_REL, function( relVal ) {
-					const regExp = new RegExp( '\\s?\\b' + relVal + '\\b\\s?', 'gi' );
+					const regExp = new RegExp( '\\b' + relVal + '\\b', 'gi' );
 					newRel = newRel.replace( regExp, '' );
 				} );
 
+				// Only trim if NEW_TAB_REL values was replaced.
+				if ( newRel !== currentRel ) {
+					newRel = newRel.trim();
+				}
+
```

Comment:
Thanks for the comment @jorgefilipecosta. This new commit basically has 2 changes.

1. Changed the regex used. I realized that `\b` is enough if we're gonna trim the spaces as well.
2. Trim the resulting rel if a NEW_TAB_REL ('noreferrer', 'noopener') was removed. This is because without this check, it's not possible to type multiple words in the Link Rel field since the whitespace is always trimmed.

Honestly, I think we can find a better solution like the removal of NEW_TAB_REL values should just happen on save. I'll try to look into it more when time permits.

@jorgefilipecosta - Let me know if your tests with this new changes are good.